### PR TITLE
Create selenium tests that capture screenshots of search results

### PR DESCRIPTION
**Pull request by aldenstpage**
Related to creativecommons/cccatalog-api#311.

@kgodey, do you have any suggestions for test search terms?

### CODE REVIEW

File: test/selenium-webdriver/search_screencaps.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+
+
+SCREEN_DIR = './screenshots'
+TARGET_URL = 'https://search.creativecommons.org'
+release = rs.get('https://api-dev.creativecommons.engineering/version').json()
```
Comment (aldenstpage) [inline test/selenium-webdriver/search_screencaps.py]: This is temporarily pointing at api-dev until the next production release of cccatalog-api.

File: .gitignore [hunk 2]
```diff
@@ -6,6 +6,7 @@ yarn-debug.log*
 yarn-error.log*
 /test/unit/coverage/
 /test/e2e/reports/
+/test/selenium/geckodriver.log
```
Comment (brenoferreira) [inline .gitignore]: I think the `/test/selenium-webdriver/screenshots/*` should be ignored too.

**Comment by kgodey** [comment]
Here are suggested search terms in addition to what you have already.

Suggested search terms reported as low quality:
- weather (reported as inappropriate)
- Andrew Jackson (reported as inappropriate)
- maps of the Atlantic
- wildlife
- reading
- Syrian women
- medicine
- health
- periodic table
- aardvark
- children playing
- nurse
- basketball team

Suggested search terms based on frequently searched terms:
- music
- computer
- technology
- car
- books
- video
- beach
- flower
- food
- tree
- people
- nature
- money
- house
- school

**Comment by aldenstpage** [comment]
@kgodey I added the search terms you listed and also updated the script to capture the second page of results.

@brenoferreira I spent a few minutes setting up Docker before realizing this isn't a good fit for it. It requires a live Firefox GUI, which means X11 needs to be running. I could probably rig up X11, X forwarding, and remote desktop to completely containerize this, but it would take a few days of work. It may be best to instead follow the README and install the two dependencies manually so you can run it natively (`geckodriver` and `pipenv`)

**Comment by brenoferreira** [comment]
@aldenstpage ah, right. Could you instead add a small script that installs the dependencies and gets the tests ready for execution? Also, add an npm script to `package.json` that runs these tests, similar to the unit tests there: https://github.com/creativecommons/cccatalog-frontend/blob/develop/package.json#L11

**Comment by aldenstpage** [comment]
@brenoferreira I've added some NPM scripts. Installation of geckodriver and pipenv are platform dependent, so I'm reluctant to include a script for it, but these particular scripts should work on your machine.

File: package.json [hunk 3]
```diff
@@ -9,6 +9,8 @@
     "dev": "webpack-dev-server --https --inline --progress --config build/webpack-dev-conf.js",
     "start": "npm run dev",
     "unit": "jest --config test/unit/jest-conf.js --coverage",
+    "qa-install": "brew install geckodriver && brew install pipenv",
```
Comment (brenoferreira) [inline package.json]: Shouldn't this include `&& cd test/selenium-webdriver && pipenv install`?

**Comment by aldenstpage** [comment]
That step is executed with `pipenv run`

File: package.json (hunk 3 above)
Comment (aldenstpage) [reply package.json]: That step is executed with `pipenv run`.

**Pull request merged**